db/sql/030_create_billable.sql

```diff
@@ -0,0 +1,124 @@
+DROP MATERIALIZED VIEW IF EXISTS billable;
+CREATE MATERIALIZED VIEW IF NOT EXISTS billable AS with
+
+-- extract useful stuff from usage events
+-- we treat both apps and services as billable "resources" so normalize the fields
+-- we normalize states to just STARTED/STOPPED because we treat consecutive STARTED to mean "update"
+events as (
+	(
+		select
+			id,
+			created_at::timestamptz as created_at,
+			(raw_message->>'app_guid') as guid,
+			(raw_message->>'app_name') as name,
+			(raw_message->>'org_guid') as org_guid,
+			(raw_message->>'space_guid') as space_guid,
+			'f4d4b95a-f55e-4593-8d54-3364c25798c4'::text as plan_guid, -- fake plan id for compute plans
+			'default-compute'::text as plan_name,                      -- fake plan name for compute plans
+			coalesce(raw_message->>'instance_count', '0')::numeric as inst_count,
```

Comment:
When do we expect this to be null? Same goes for `memory_in_mb_per_instance` on the line below.

Reply:
It's not expected to be null ... but there's nothing currently _stopping_ it from being null and we want that column to always be not null. Bit overprotective maybe.

`(select (NULL + 1) is null)` => `null`  ...which would be a bad thing for calcuations.

We could get the same behaviour by adding an index/trigger to validate the incoming events data, but decided that handling the error cases of a bad incoming event was much harder than filtering the data.